smpplib/client.py

```diff
@@ -283,7 +283,7 @@ def read_once(self, ignore_error_codes=None):
                 self.send_pdu(p)
                 return
 
-            if p.is_error():
+            if p.command != 'submit_sm_resp' and p.is_error():
```

Comment:
This is too specific to the particular type of PDU and breaks the general flow. I rather like @podshumok's suggestion to add `raise_for_status` or something like that.